### PR TITLE
Update waaw

### PR DESCRIPTION
**Pull request by mbebe**

### CODE REVIEW

File: lib/resolveurl/plugins/waaw.py [hunk 1]
```diff
@@ -22,146 +24,27 @@ class WaawResolver(ResolveUrl):
...
-
+    
     def __init__(self):
         self.net = common.Net()
-
+    
     def get_media_url(self, host, media_id):
         web_url = self.get_url(host, media_id)
         headers = {'User-Agent': common.FF_USER_AGENT,
-                   'Referer': 'https://waaw.tv/watch_video.php?v=%s&post=1' % media_id}
```
Comment (Gujal00) [inline lib/resolveurl/plugins/waaw.py:32]: may still need `'Referer': web_url` in headers, please check

File: lib/resolveurl/plugins/waaw.py [hunk 2]
```diff
@@ -22,146 +24,27 @@ class WaawResolver(ResolveUrl):
...
-        return "".join(v7)
-
+                   'x-requested-with': 'XMLHttpRequest'}
+        params = (
+        	('ver', '0'),
+        	('secure', '0'),
+        	('adb', '0/'),
+        	('v', media_id),
+        	('token', ''),
+        	('gt', ''),)
```
Comment (Gujal00) [inline lib/resolveurl/plugins/waaw.py]: redundant , before closing )

File: lib/resolveurl/plugins/waaw.py [hunk 3]
```diff
@@ -22,146 +24,27 @@ class WaawResolver(ResolveUrl):
...
+        	('ver', '0'),
+        	('secure', '0'),
+        	('adb', '0/'),
+        	('v', media_id),
+        	('token', ''),
+        	('gt', ''),)	   
+        params = urllib.urlencode(params)
+        rurl = "https://hqq.tv/player/get_md5.php?" + params  
+        str_url = self.net.http_HEAD(rurl, headers=headers).get_url()
+        return str_url+'.mp4.m3u8' + helpers.append_headers(headers)
```
Comment (Gujal00) [inline lib/resolveurl/plugins/waaw.py:44]: needs `headers.pop('x-requested-with')` before this line. Please check

**Review by Gujal00 — CHANGES_REQUESTED** [review]

File: lib/resolveurl/plugins/waaw.py (hunk 1 above)
Comment (mbebe) [reply lib/resolveurl/plugins/waaw.py:32]: Playing is correct without referer.

File: lib/resolveurl/plugins/waaw.py (hunk 3 above)
Comment (mbebe) [reply lib/resolveurl/plugins/waaw.py:44]: It's not necessary.

File: lib/resolveurl/plugins/waaw.py (hunk 1 above)
Comment (Gujal00) [reply lib/resolveurl/plugins/waaw.py:32]: okay in which case why is web_url even defined, if it is not going to be used

File: lib/resolveurl/plugins/waaw.py (hunk 3 above)
Comment (Gujal00) [reply lib/resolveurl/plugins/waaw.py:44]: It is good practice to not send unnecessary headers when requesting stream url. "X-Requested-With" generally is never used with mp4 or m3u8 urls

Comment (Gujal00) [reply lib/resolveurl/plugins/waaw.py:44]: please include `headers.pop('x-requested-with')` before the return statement

**Review by Gujal00 — APPROVED** [review]

**Pull request merged**